### PR TITLE
Updated configuration for Docker-based devstack

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-6570

### CODE REVIEW

**Comment by clintonb** [comment]
@rlucioni @davec-edx @jmbowman please review
FYI @edx/devops

File: Makefile [hunk 1]
```diff
@@ -14,6 +15,7 @@ help:
 	@echo "  help                       display this help message"
 	@echo "  html_coverage              generate and view HTML coverage report"
 	@echo "  migrate                    apply database migrations"
+	@echo "  production-requirements    install requirements for production"
```
Comment (davec-edx) [inline Makefile]: use of - in target name...other targets use _

File: Makefile [hunk 2]
```diff
@@ -32,7 +34,10 @@ clean:
 	coverage erase
 
 requirements:
-	pip install -r requirements/local.txt
+	pip install -r requirements/local.txt --exists-action w
+
+production-requirements:
```
Comment (davec-edx) [inline Makefile]: - v _

File: programs/settings/base.py [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import os
+import platform
+from logging.handlers import SysLogHandler
 from os.path import join, abspath, dirname
```
Comment (davec-edx) [inline programs/settings/base.py]: I'm thinking we should have a common settings for IDAs or configuration service.  

Do we have a way of dynamically changing things, such as the logging level?

File: programs/settings/base.py [hunk 4]
```diff
@@ -251,3 +253,76 @@
...
+            'propagate': True,
+            'level': 'WARNING'
+        },
+        '': {
+            'handlers': ['console', 'local'],
+            'level': 'DEBUG',
+            'propagate': False
+        },
+    }
+}
```
Comment (davec-edx) [inline programs/settings/base.py]: Consider tabular layout.

File: programs/settings/base.py (hunk 3 above)
Comment (rlucioni) [inline programs/settings/base.py]: @davec-edx @jmbowman's work may help address this. We don't have a good answer right now.

File: requirements/production.txt [hunk 5]
```diff
@@ -4,5 +4,6 @@
 gevent==1.0.2
 gunicorn==19.3.0
 MySQL-python==1.2.5
+newrelic==2.72.1.53
 PyYAML==3.11
 nodeenv==0.13.6
```
Comment (rlucioni) [inline requirements/production.txt]: Nodeenv is still here. Shouldn't we be consistent about how we choose to install it?

File: programs/settings/base.py (hunk 4 above)
Comment (rlucioni) [inline programs/settings/base.py]: @davec-edx let's address logging-related changes separately. As it stands, @clintonb's just moving what was already here.

File: programs/settings/base.py (hunk 3 above)
Comment (clintonb) [inline programs/settings/base.py]: I agree that we should have common settings. At some point, I will repurpose https://github.com/edx/edx-django-extensions for this need.

File: requirements/production.txt (hunk 5 above)
Comment (clintonb) [inline requirements/production.txt]: Yes, it should go away here.

**Pull request merged**